.github/workflows/deploy.yml

```diff
@@ -0,0 +1,131 @@
+name: Deploy
+
+on:
+  workflow_dispatch:
+    inputs:
+      platforms:
+        description: 'Platforms to build:'
+        default: 'linux_arm32 linux_arm64 linux_x64 macos windows_x64 windows_portable' # TODO: backend
+        required: true
+      release_type:
+        description: 'Release type: test,alpha,beta,rc,stable'
+        default: 'test'
+        required: true
+      create_tag:
+        description: 'Create tag: on,off (if on, release type must be alpha,beta,rc,stable)'
+        default: 'off'
+        required: false
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  get_tag_info:
+    name: Get tag info
+    runs-on: ubuntu-latest
+    outputs:
+      version_label: ${{ steps.make_tag_info.outputs.VERSION_LABEL }}
+      tag_name: ${{ steps.make_tag_info.outputs.TAG_NAME }}
+      release_name: ${{ steps.make_tag_info.outputs.RELEASE_NAME }}
+    steps:
+    - name: Clone repo with history
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0 # entire history for all branches and tags
+    - id: make_tag_info
+      name: Make tag info
+      env:
+        RELEASE_TYPE: ${{ inputs.release_type }}
+        CREATE_TAG: ${{ inputs.create_tag }}
+      run: |
+        git tag --list | tail -n 20
+        echo ==========
+        buildscripts/ci/release/make_tag_name.sh | tee -a "${GITHUB_OUTPUT}"
+        echo ==========
+        echo "${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/commit/${GITHUB_SHA}"
+        echo ==========
+        if [[ "${CREATE_TAG}" == 'on' ]]; then
+          echo "Tag will be created."
+        else
+          echo "Tag will not be created."
+        fi
+
+  build:
+    name: Build
+    needs: get_tag_info
+    uses: ./.github/workflows/build_all.yml
+    with:
+      platforms: ${{ inputs.platforms }}
+      build_mode: ${{ inputs.release_type == 'stable' && 'stable' || 'testing' }}
+      publish: ${{ inputs.create_tag == 'on' && 'on' || 'off' }}
+      sentry_project: ${{ inputs.mode == 'stable' && 'mu4' || 'sandbox' }}
+      version_label: ${{ needs.get_tag_info.outputs.version_label }}
+
+  update_learn_playlists:
+    name: 'Update Home→Learn playlists'
+    needs: build
+    uses: ./.github/workflows/update_learn_playlists.yml
+    with:
+      mode: ${{ inputs.mode }}
+      environment: production # requires approval
+
+  create_release:
+    name: 'Create release: ${{ needs.get_tag_info.outputs.tag_name }}'
+    needs:
+      - get_tag_info # to access outputs
+      - update_learn_playlists
+    if: ${{ ! cancelled() && inputs.create_tag == 'on' }}
+    runs-on: ubuntu-latest
+    environment:
+      name: production # requires approval
+      url: ${{ github.server_url }}/${{ github.repository }}/releases/tag/${{ needs.get_tag_info.outputs.tag_name }}
+    steps:
+    - name: Clone repository
+      uses: actions/checkout@v4
+    - name: Download and extract artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: build.artifact
```

Comment:
`collate_release_binaries` has <code>build.artifact<I>s</I></code>. Shouldn't they be in sync?